powerquery-docs/data-privacy-firewall.md

```diff
@@ -154,9 +154,10 @@ Here's a high-level summary of the partitioning logic.
       * Note that "removing" a partition effectively includes it in whatever other partitions reference it.
       * Trimming parameter partitions allows parameter references used within data source function calls (for example, `Web.Contents(myUrl)`) to work, instead of throwing "partition can't reference data sources and other steps" errors.
     * Grouping (Static)
-      * Partitions are merged, while maintaining separation between:
-        * Partitions in different queries
-        * Partitions that reference other partitions vs. those that don't
+      * Partitions are merged. In the resulting merged partitions, the following will be separate:
```

Comment:
This summary does not seem to incorporate the following rule:
> •	Partitions are then merged in bottom-up dependency order. Partitions A and B will be merged if the following conditions apply: 
o	B is not the top-level result returned by the query.
o      [....]

Arguably, a goal of this article is for the reader to be able to figure out how PQ will partition a query, so it seems this rule should be factored in.

However, based on my last exchange with @ehrenMSFT, not sure that I understand this rule all the way, so am not sure how to summarize it into a bullet.

@ehrenMSFT, can you help me understand the rule? Thanks!

Reply:
Revised to incorporate this rule. What do you think, @ehrenMSFT?

Reply:
This is incorrect: "The query's final output partition will not have other partitions merged into it."

It's actually the other way around. The final output partition won't be merged into another partition.

Reply:
This is incorrect: "The query's final output partition will not have other partitions merged into it."

It's slighly more subtle: the output partition can merge with partitions it references, but not with partitions that reference it.

Reply:
Thanks, @ehrenMSFT! Any better now? :-)

Reply:
I think the special case is kind of confusing. What is a "partition boundary"? What is a query's "return statement"?

How about this?

Partitions are merged. The goal is to merge the partitions in bottom-up dependency order so that in the resulting merged partitions, the following will be separate:
        * Partitions in different queries
        * Partitions that don't reference other partitions (and are thus allowed to access a data source)
        * Partitions that reference other partitions (and are thus prohibited from accessing a data source)

Reply:
Thanks, @ehrenMSFT. How does it look now?

Reply:
That looks good to me.